be/src/vec/columns/column_struct.cpp

```diff
@@ -185,6 +185,27 @@ const char* ColumnStruct::deserialize_and_insert_from_arena(const char* pos) {
     return pos;
 }
 
+int ColumnStruct::compare_at(size_t n, size_t m, const IColumn& rhs_,
+                             int nan_direction_hint) const {
+    const ColumnStruct& rhs = assert_cast<const ColumnStruct&>(rhs_);
+
+    const size_t lhs_tuple_size = columns.size();
+    const size_t rhs_tuple_size = rhs.tuple_size();
+
+    if (rhs_tuple_size < lhs_tuple_size) {
```

Comment:
is it correct to compare tuple size first? why not compare element first?

Reply:
I think it's correct, because struct is combination of multi columns, only  tuple size is same, we can compare at elem column. other is self column and offset column.